### PR TITLE
Feat/fuseswap ui

### PR DESCRIPTION
**Pull request by leonprou**
- Fuse swap changes
- Fuse bridge

@mul53 ❤️

### CODE REVIEW

File: src/components/SearchModal/CurrencySearchModal.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Currency } from '@uniswap/sdk'
+import { Currency } from 'uniswap-fuse-sdk'
```
Comment (leonprou) [inline src/components/SearchModal/CurrencySearchModal.tsx:1]: @mul53 can you give link to uniswap-fuse-sdk

File: src/pages/AddLiquidity/index.tsx [hunk 2]
```diff
@@ -137,13 +137,12 @@ export default function AddLiquidity({
...
 
-    let estimate,
-      method: (...args: any) => Promise<TransactionResponse>,
+    let method: (...args: any) => Promise<TransactionResponse>,
       args: Array<string | string[] | number>,
       value: BigNumber | null
     if (currencyA === ETHER || currencyB === ETHER) {
       const tokenBIsETH = currencyB === ETHER
-      estimate = router.estimateGas.addLiquidityETH
+      // estimate = router.estimateGas.addLiquidityETH
```
Comment (leonprou) [inline src/pages/AddLiquidity/index.tsx:145]: why this is commented?

File: package.json [hunk 3]
```diff
@@ -105,5 +104,9 @@
       "last 1 safari version"
     ]
   },
-  "license": "GPL-3.0-or-later"
+  "license": "GPL-3.0-or-later",
+  "dependencies": {
+    "fuse-default-token-list": "^1.5.4",
```
Comment (leonprou) [inline package.json:109]: so you added these packages to NPM?

File: package.json [hunk 4]
```diff
@@ -105,5 +104,9 @@
...
+    "uniswap-fuse-sdk": "^3.0.3-a4"
```
Comment (leonprou) [inline package.json:110]: can you add link to this repo?

**Review by leonprou — COMMENTED** [review]
@mul53 good job, not done yet but some coments for now

File: src/components/SearchModal/CurrencySearchModal.tsx (hunk 1 above)
Comment (mul53) [reply src/components/SearchModal/CurrencySearchModal.tsx:1]: here is the [code](https://github.com/mul53/uniswap-sdk)

File: src/pages/AddLiquidity/index.tsx (hunk 2 above)
Comment (mul53) [reply src/pages/AddLiquidity/index.tsx:145]: it was causing some errors, i can look further into it

File: package.json (hunk 3 above)
Comment (mul53) [reply package.json:109]: yes i added those packages to npm

File: package.json (hunk 4 above)
Comment (mul53) [reply package.json:110]: here is the repo for the default token list https://github.com/mul53/default-token-list

File: package.json (hunk 3 above)
Comment (leonprou) [reply package.json:109]: got you. So after I'll publish this on `npm` https://github.com/fuseio/fuse-token-list, we can switch using this list?

File: package.json (hunk 4 above)
Comment (leonprou) [reply package.json:110]: was looking for a repo for uniswap-fuse-sdk :)

File: package.json (hunk 3 above)
Comment (mul53) [reply package.json:109]: yes we can switch no problem

File: package.json (hunk 4 above)
Comment (mul53) [reply package.json:110]: here it is https://github.com/mul53/uniswap-sdk

Comment (leonprou) [reply package.json:110]: thanks 👍 
can you move the dependencies to devDependecnies like uniswap team is doing?

Comment (mul53) [reply package.json]: Done

File: src/pages/AddLiquidity/index.tsx (hunk 2 above)
Comment (leonprou) [reply src/pages/AddLiquidity/index.tsx:145]: better to check this.But less important than the UI fixes in telegram

File: src/components/Button/index.tsx [hunk 5]
```diff
@@ -110,22 +110,22 @@ export const ButtonGray = styled(Base)`
 `
 
 export const ButtonSecondary = styled(Base)`
-  background-color: ${({ theme }) => theme.primary5};
-  color: ${({ theme }) => theme.primaryText1};
+  background-color: ${({ theme }) => theme.bg7};
+  color: #d4eeff;
```
Comment (leonprou) [inline src/components/Button/index.tsx:114]: Can you try themes colours here?

File: src/components/CurrencyLogo/index.tsx [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
+import FuseLogo from '../../assets/images/fuse-logo.png'
 import EthereumLogo from '../../assets/images/ethereum-logo.png'
 import useHttpLocations from '../../hooks/useHttpLocations'
 import { WrappedTokenInfo } from '../../state/lists/hooks'
 import Logo from '../Logo'
+import { useActiveWeb3React } from '../../hooks'
 
 const getTokenLogoURL = (address: string) =>
-  `https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/${address}/logo.png`
+  `https://raw.githubusercontent.com/mul53/default-token-list/master/assets/${address}/logo.png`
```
Comment (leonprou) [inline src/components/CurrencyLogo/index.tsx:13]: Need to switch this for using our fuseio repo. If something missing there please open a PR

File: src/components/Header/index.tsx [hunk 7]
```diff
@@ -127,30 +109,27 @@ const BalanceText = styled(Text)`
 `
 
 const NETWORK_LABELS: { [chainId in ChainId]: string | null } = {
-  [ChainId.MAINNET]: null,
+  [ChainId.MAINNET]: 'Mainnet',
```
Comment (leonprou) [inline src/components/Header/index.tsx:112]: please rename `Mainnet` to `Ethereum`?

**Review by leonprou — COMMENTED** [review]
Looks good 💯 
some small comments given

**Comment by leonprou** [comment]
@mul53 can you also make a PR to the uniswap sdk for to here https://github.com/fuseio/fuseswap-sdk ?

File: src/constants/lists.ts [hunk 8]
```diff
@@ -1,20 +1,5 @@
 // the Uniswap Default token list lives here
-export const DEFAULT_TOKEN_LIST_URL = 'tokens.uniswap.eth'
+export const DEFAULT_TOKEN_LIST_URL =
```
Comment (leonprou) [inline src/constants/lists.ts:2]: please change to https://www.npmjs.com/package/@fuseio/fuse-swap-default-token-list

File: src/components/CurrencyLogo/index.tsx (hunk 6 above)
Comment (leonprou) [reply src/components/CurrencyLogo/index.tsx:13]: https://www.npmjs.com/package/@fuseio/fuse-swap-default-token-list

Comment (mul53) [reply src/components/CurrencyLogo/index.tsx:13]: why don't we use trust wallet instead? I can open a pr there

File: src/components/Button/index.tsx (hunk 5 above)
Comment (mul53) [reply src/components/Button/index.tsx:114]: No problem

File: src/components/Header/index.tsx (hunk 7 above)
Comment (mul53) [reply src/components/Header/index.tsx:112]: Done

File: src/components/AccountDetails/index.tsx [hunk 9]
```diff
@@ -4,7 +4,7 @@ import styled, { ThemeContext } from 'styled-components'
 import { useActiveWeb3React } from '../../hooks'
 import { AppDispatch } from '../../state'
 import { clearAllTransactions } from '../../state/transactions/actions'
-import { shortenAddress } from '../../utils'
+import { shortenAddress, getEtherscanLinkText } from '../../utils'
```
[the remark this answers is not in the document]
Comment (mul53) [reply src/components/AccountDetails/index.tsx:7]: done

File: src/connectors/index.ts [hunk 10]
```diff
@@ -44,8 +45,9 @@ export const getNetworkLibraryByChain = (chainId: number) => {
     })
     return new Web3Provider(network.provider as any)
   } else if (chainId === ChainId.FUSE) {
+    const networkUrl = HOME_BRIDGE_CHAIN === ChainId.ROPSTEN ? ROPSTEN_NETWORK_URL : MAINNET_NETWORK_URL
```
Comment (leonprou) [inline src/connectors/index.ts:48]: `HOME` is always the fuse :)
foreign it ropsten or main

File: .env [hunk 11]
```diff
@@ -1,3 +1,5 @@
 REACT_APP_CHAIN_ID="122"
 REACT_APP_NETWORK_URL="https://rpc.fuse.io/"
+REACT_APP_ROPSTEN_NETWORK_URL=https://ropsten.infura.io/v3/eab5c3b10b2b409e998a7528d7d81275
```
Comment (leonprou) [inline .env:3]: that's your infura account, right?
can you please change to `2ba4786ec53f4b4aadd7ef61f37e5080`

File: src/components/SearchModal/CurrencySearchModal.tsx (hunk 1 above)
Comment (leonprou) [reply src/components/SearchModal/CurrencySearchModal.tsx:1]: can you change this once again to `@fuseio/fuse-swap-sdk`  :) ?

Comment (leonprou) [reply src/components/SearchModal/CurrencySearchModal.tsx:1]: all references ofc

File: package.json (hunk 3 above)
Comment (leonprou) [reply package.json]: please change again to `@fuseio/fuse-swap-default-token-list`  :)

File: .env (hunk 11 above)
Comment (mul53) [reply .env:3]: Yes, I will change that

File: src/connectors/index.ts (hunk 10 above)
Comment (mul53) [reply src/connectors/index.ts:48]: @leonprou I added a conditional for Lior he wanted to test the functionality with the home bridge set to ropsten, should i change that?

File: cypress/support/commands.js [hunk 12]
```diff
@@ -68,15 +73,39 @@ class CustomizedBridge extends _Eip1193Bridge {
 }
 
 // sets up the injected provider to be a mock ethereum provider with the given mnemonic/index
+// Cypress.Commands.overwrite('visit', (original, url, options) => {
```
Comment (leonprou) [inline cypress/support/commands.js:76]: please remove the commented code

File: cypress/support/commands.js [hunk 13]
```diff
@@ -68,15 +73,39 @@ class CustomizedBridge extends _Eip1193Bridge {
...
     onBeforeLoad(win) {
       options && options.onBeforeLoad && options.onBeforeLoad(win)
       win.localStorage.clear()
-      const provider = new JsonRpcProvider('https://rinkeby.infura.io/v3/4bf032f2d38a4ed6bb975b80d6340847', 4)
+
+      let args, chainId
+
+      if (options && options.networkName == 'ropsten') {
+        chainId = '0x3'
+        args = ['https://ropsten.infura.io/v3/eab5c3b10b2b409e998a7528d7d81275', 3]
```
Comment (leonprou) [inline cypress/support/commands.js:100]: better use the env here

**Review by leonprou — COMMENTED** [review]
changing looking good!
added some more small comments.
also need to sort the home/foreign confusion

File: src/connectors/index.ts (hunk 10 above)
Comment (leonprou) [reply src/connectors/index.ts:48]: there's home and foreign chains. Home is always fuse, while foreign is Ethereum or Ropsten (for tests).
so checking `HOME_BRIDGE_CHAIN === ChainId.ROPSTEN` doesn't make any sense. 
I see that came from env file. the env is
```
REACT_APP_CHAIN_ID="122"
REACT_APP_HOME_BRIDGE_CHAIN=1
```
`REACT_APP_CHAIN_ID` is represents the home chain
and `REACT_APP_HOME_BRIDGE_CHAIN` is Ethereum, so it's a foreign network actually
so should rename `REACT_APP_HOME_BRIDGE_CHAIN` to `REACT_APP_FOREIGN_BRIDGE_CHAIN` (and all the derived variables)

Comment (mul53) [reply src/connectors/index.ts:48]: @leonprou  I have renamed it, that name was causing the confusion, i will work better on naming

Comment (leonprou) [reply src/connectors/index.ts:48]: cool, can you push the latest PR fixes?

Comment (mul53) [reply src/connectors/index.ts]: I have added the latest PR fixes

**Comment by mul53** [comment]
@leonprou Added the changes

File: cypress/support/commands.js [hunk 14]
```diff
@@ -97,10 +85,10 @@ Cypress.Commands.overwrite('visit', (original, url, options) => {
 
       if (options && options.networkName == 'ropsten') {
         chainId = '0x3'
-        args = ['https://ropsten.infura.io/v3/eab5c3b10b2b409e998a7528d7d81275', 3]
+        args = [process.env.REACT_APP_ROPSTEN_NETWORK_URL, 3]
       } else {
         chainId = '0x7A'
-        args = ['https://rpc.fuse.io', 122]
+        args = [process.env.REACT_APP_CHAIN_ID, 122]
```
Comment (leonprou) [inline cypress/support/commands.js:91]: @mul53 should  [REACT_APP_NETWORK_URL, REACT_APP_CHAIN_ID]

**Pull request merged**